src/main/java/com/cedrus/aeolion/kafkaspringpong/dao/SpringPongDaoImpl.java

```diff
@@ -0,0 +1,46 @@
+package com.cedrus.aeolion.kafkaspringpong.dao;
+
+import com.cedrus.aeolion.kafkaspringpong.model.SpringPongBall;
+import com.cedrus.aeolion.kafkaspringpong.util.SpringPongRowMapper;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+import javax.sql.DataSource;
+import java.util.List;
+
+public class SpringPongDaoImpl implements SpringPongDao {
+    private final JdbcTemplate jdbcTemplate;
+    private final SpringPongRowMapper springPongRowMapper;
+
+    @Autowired
+    public SpringPongDaoImpl(DataSource ds, SpringPongRowMapper springPongRowMapper) {
+        jdbcTemplate = new JdbcTemplate(ds);
+        this.springPongRowMapper = springPongRowMapper;
+    }
+
+    public SpringPongBall getBallById(String id) {
+        final String sql = "SELECT * FROM spring_pong WHERE id = " + id;
```

Comment:
Not a good pattern.  This is an invitation for SQL injection.  Please use a parameterized invocation.  Talk to Pat about how he did it. You'll see that it works differently.  

https://xkcd.com/327/